windows/wrapper/impl_webrtc_AudioDeviceWasapi.h

```diff
@@ -0,0 +1,520 @@
+
+      bool CheckBuiltInCaptureCapability(winrt::Windows::Media::Effects::AudioEffectType) const;
+      bool CheckBuiltInRenderCapability(winrt::Windows::Media::Effects::AudioEffectType) const;
+
+      void _SetThreadName(DWORD dwThreadID, LPCSTR szThreadName);
+      void _Lock() { _critSect.Enter(); }
+      void _UnLock() { _critSect.Leave(); }
+
+   private:
+      int32_t InitRecordingInternal();
+      int32_t StartRecordingInternal();
+      int32_t StopRecordingInternal();
+
+      int32_t InitPlayoutInternal();
+      int32_t StartPlayoutInternal();
+      int32_t StopPlayoutInternal();
+
+      int SetBoolProperty(IPropertyStore* ptrPS,
+                          REFPROPERTYKEY key,
+                          VARIANT_BOOL value);
+
+      int SetVtI4Property(IPropertyStore* ptrPS,
+                          REFPROPERTYKEY key,
+                          LONG value);
+
+      int32_t _EnumerateEndpointDevicesAll();
+      void _TraceCOMError(HRESULT hr) const;
+      void _Get44kHzDrift();
+
+      int32_t _RefreshDeviceList(DeviceClass cls);
+      int16_t _DeviceListCount(DeviceClass cls);
+
+      winrt::hstring  _GetDefaultDeviceName(DeviceClass cls);
+      winrt::hstring  _GetListDeviceName(DeviceClass cls, int index);
+      winrt::hstring  _GetDeviceName(DeviceInformation const &device);
+
+      winrt::hstring  _GetListDeviceID(DeviceClass cls, int index);
+      winrt::hstring  _GetDefaultDeviceID(DeviceClass cls);
+      winrt::hstring  _GetDeviceID(DeviceInformation const &device);
+
+      DeviceInformation _GetDefaultDevice(DeviceClass cls, AudioDeviceRole role);
+      DeviceInformation _GetListDevice(DeviceClass cls, int index);
+      DeviceInformation _GetListDevice(DeviceClass cls, winrt::hstring const &deviceId);
+
+      void _InitializeAudioDeviceIn(winrt::hstring const &deviceId);
+      void _InitializeAudioDeviceOut(winrt::hstring const &deviceId);
+
+      // Surround system support
+      bool ShouldUpmix();
+      WAVEFORMATEX* GenerateMixFormatForMediaEngine(
+        WAVEFORMATEX* actualMixFormat);
+      WAVEFORMATPCMEX* GeneratePCMMixFormat(WAVEFORMATEX* actualMixFormat);
+      void Upmix(int16_t *inSamples, uint32_t numberOfFrames,
+        int16_t *outSamples, uint32_t inChannels, uint32_t outChannels);
+      void UpmixAndConvert(int16_t *inSamples, uint32_t numberOfFrames,
+        float *outSamples, uint32_t inChannels, uint32_t outChannels);
+
+   private:
+      void DefaultAudioCaptureDeviceChanged(
+        DefaultAudioCaptureDeviceChangedEventArgs const &args);
+      void DefaultAudioRenderDeviceChanged(
+        DefaultAudioRenderDeviceChangedEventArgs const &args);
+
+   private:
+     AudioDeviceWasapiWeakPtr thisWeak_;
+     mutable zsLib::RecursiveLock lock_;
+
+     IAudioDeviceWasapiDelegateSubscriptions subscriptions_;
+     IAudioDeviceWasapiSubscriptionPtr defaultSubscription_;
+
+     std::string id_;
+
+   private:
+      ScopedCOMInitializer                    _comInit;
```

Comment:
variables should have _ at the end not the start

Reply:
Fixed on Sergej/20181206-m71-python